ui/src/app/pages/artifacts/artifacts.tsx

```diff
@@ -295,13 +295,15 @@ export class ArtifactsPage extends PageComponent<ArtifactsPageProps, ArtifactsPa
                 const artifactLocation: string = this.linkTo(`/artifacts/${ encodeURIComponent(groupId) }/${ encodeURIComponent(metaData.id) }`);
                 Services.getLoggerService().info("[ArtifactsPage] Artifact successfully uploaded.  Redirecting to details: ", artifactLocation);
                 this.navigateTo(artifactLocation)();
+                this.setMultiState({uploadFormData: null, isUploadFormValid: false});
```

Comment:
This one I think should not be needed, unless the navigation were to fail for some reason.  No?  The line before this one should navigate the user to a new page.

Reply:
I think this one is just in case

Reply:
Can you remove this one and add a comment on the other one about why we're doing it?  Specifically about the fact that the navigation in that case (and only that case) doesn't send the user to a new page, it just loads the current page with a new version (the version that was just uploaded).

The reason is that right now the UI is relying on the page state being reset when navigating.  So the upload artifact version use-case is the exceptional case and is the only one that should reset the state after navigating.  

If we want we can move the page state initialization to a different part of the page lifecycle that will cover the "navigating to the same page" use-case.  But I would suggest that's a separate PR.

Reply:
I removed both, I just realized the only cleanups needed are in the error catch functions, because when there is an error we only show a modal, and there the state is 100% kept ( so cleanup is needed) for the other scenarios I haven't been able to reproduce any issue now

Reply:
Perfect thanks.